packages/core/src/entity/EntityAssigner.ts

```
@@ -124,16 +124,19 @@ export class EntityAssigner {
     const Embeddable = prop.embeddable;
     const propName = prop.embedded ? prop.embedded[1] : prop.name;
     entity[propName] = options.mergeObjects ? entity[propName] || Object.create(Embeddable.prototype) : Object.create(Embeddable.prototype);
-
-    Object.keys(value).forEach(key => {
-      const childProp = prop.embeddedProps[key];
-
-      if (childProp && childProp.reference === ReferenceType.EMBEDDED) {
-        return EntityAssigner.assignEmbeddable(entity[propName], value[key], childProp, em, options);
-      }
-
-      entity[propName][key] = value[key];
-    });
+    if (prop.nullable && value === null) {
+      entity[propName] = null;
+    } else if (typeof value === 'object') {
```

i'd rather keep this as simple `else` branch (as noted above, if we return early, there is no need for new branching at all), or add a validation if the value is not object. doing it like this, it would silently ignore badly formed data - we should rather fail in such case.

also this will definitely need some test cases to keep the 100% branch coverage

---

@B4nan thank you very much for your quick response! 
I'll tell you the reasons that made me do it this way:
- In the first if I understand that null allows to delete from the database, so I assign null but only if the property is nullable, to prevent deleting something if it is not configured for it.
- In the second if, the object type check allows to discard two cases, on the one hand null which is not nullable and on the other hand undefined, which is an ignored value and respects the previous value.

These are the reasons, do you think it is preferable to make the changes you are talking about?

Thanks a lot!

---

As I said, we should not ignore malformed input data, we should throw instead. e.g. we could throw when user tries to set null/undefined and the property is not nullable, but ignoring the request to set a property to null, just because it is not nullable, that sounds wrong to me.

---

You are absolutely right! I have updated it